src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/RecommendCaseInsensitiveStringComparison.Fixer.cs

```diff
@@ -0,0 +1,201 @@
+            {
+                return;
+            }
+
+            SemanticModel model = await doc.GetSemanticModelAsync(ct).ConfigureAwait(false);
+
+            if (model.GetOperation(node, ct) is not IInvocationOperation invocation)
+            {
+                return;
+            }
+
+            if (model.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemStringComparison)
+                is not INamedTypeSymbol stringComparisonType)
+            {
+                return;
+            }
+
+            if (model.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemStringComparer)
+                is not INamedTypeSymbol stringComparerType)
+            {
+                return;
+            }
+
+            // Only diagnose overloads that take one string argument:
+            // Contains(string), IndexOf(string), StartsWith(string), CompareTo(string)
+            if (invocation.Arguments.Count() != 1)
+            {
+                return;
+            }
+
+            // There should be a child invocation on the left side (instance) of the large invocation
+            // If the large invocation is "a.ToLower().Contains(b)"
+            // Its instance is "a.ToLower()", the child invocation
+            if (invocation.Instance is not IInvocationOperation instanceOperation)
+            {
+                return;
+            }
+
+            string caseChangingMethodName;
+            switch (instanceOperation.TargetMethod.Name)
+            {
+                case RecommendCaseInsensitiveStringComparisonAnalyzer.StringToLowerMethodName or RecommendCaseInsensitiveStringComparisonAnalyzer.StringToUpperMethodName:
+                    caseChangingMethodName = RecommendCaseInsensitiveStringComparisonAnalyzer.StringComparisonCurrentCultureIgnoreCaseName;
+                    break;
+                case RecommendCaseInsensitiveStringComparisonAnalyzer.StringToLowerInvariantMethodName or RecommendCaseInsensitiveStringComparisonAnalyzer.StringToUpperInvariantMethodName:
+                    caseChangingMethodName = RecommendCaseInsensitiveStringComparisonAnalyzer.StringComparisonInvariantCultureIgnoreCaseName;
+                    break;
+                default:
+                    return;
+            }
+
+            Task<Document> createChangedDocument(CancellationToken _) => FixInvocationAsync(doc, root,
+                invocation, instanceOperation,
+                stringComparisonType, stringComparerType,
+                invocation.TargetMethod.Name, caseChangingMethodName, ct);
+
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    title: MicrosoftNetCoreAnalyzersResources.RecommendCaseInsensitiveStringComparisonTitle,
+                    createChangedDocument,
+                    equivalenceKey: MicrosoftNetCoreAnalyzersResources.RecommendCaseInsensitiveStringComparisonTitle + invocation.TargetMethod.Name + caseChangingMethodName),
```

Comment:
Wouldn't this require few FixAll passes to actually fix all diagnostics?

Reply:
Can you please elaborate? I'm not very familiar with this.

Reply:
I see. So I guess I need to change the equivalenceKey string to something that would work better with FixAll, correct? If that's the case, what is the recommended format to use for such string?

Reply:
`nameof(MicrosoftNetCoreAnalyzersResources.RecommendCaseInsensitiveStringComparerStringComparisonCodeFixTitle)` will work well.